dpnp/dpnp_iface_mathematical.py

```diff
@@ -1850,7 +1856,14 @@ def sum(
 
             if sum:
                 sum(input, output, []).wait()
-                return dpnp_array._create_from_usm_ndarray(output)
+                result = dpnp_array._create_from_usm_ndarray(output)
+
+                if keepdims:
+                    result = result.reshape((1,) + output.shape)
+                    if axis == (1,):
+                        result = result.T
```

Comment:
```suggestion
                    res_sh = (1,) + output.shape if axis == (0,) else output.shape +(1,)
                    result = result.reshape(res_sh)
```

Reply:
Yesp quite mess. Probably it might be better formatted, like explicitly:
```python
if axis == (0,):
    res_sh = (1,) + output.shape
else:
    res_sh = output.shape + (1,)
```
I just proposed to get rid of excess call of _dpnp.ndarray.T_ method.

Reply:
`dpnp.ndarray.T` method doesn't do anything anything bad. It returns new view with changed strides. Since second dimension is 1 the view keeps to be contiguous.
I'm preferring `dpnp.ndarray.T` method because it is less verbose than `if-else`.

But if you don't like `dpnp.ndarray.T` method I can modify it to `if-else`

Reply:
Yes, `dpnp.ndarray.T` methods is quite light, but still produces a number of consequence calls, like `dpnp.ndarray.get_array()`, `dpctl.tensor.usm_ndarray.T` and `dpnp.ndarray._create_from_usm_ndarray` with unpacking and packing dpnp array back from dpctl `usm_ndarray`.

While from another way, it just needs to put (1,) at right place within `if-else`. Thus I'd say that `if-else` looks better from my view.

Reply:
Done